File: assignments/a1/q8/Python/Tkn.py
#Tkn:top#
#Tkn:import#

# <rule>:Tkn ::= <TOKEN>
class Tkn(Rule): #Tkn:class#

    className = "Tkn"
    ruleString = "<rule>:Tkn ::= <TOKEN>"
    token = None

    def __init__(token):
        #Tkn:init#
        self.token = token

#Tkn#

File: assignments/a1/q8/Java/Tkn.java
//Tkn:top//
//Tkn:import//
import java.util.*;

// <rule>:Tkn ::= <TOKEN>
public class Tkn extends Rule /*Tkn:class*/ {

    public static final String $className = "Tkn";
    public static final String $ruleString =
        "<rule>:Tkn ::= <TOKEN>";

    public Token token;

    public Tkn(Token token) {
//Tkn:init//
        this.token = token;
    }

    public static Tkn parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<rule>:Tkn", scn$.lno);
        Token token = scn$.match(Token.Match.TOKEN, trace$);
        return new Tkn(token);
    }

//Tkn//
}
